packages/core-ethereum/crates/core-ethereum-types/src/chain_events.rs

```diff
@@ -0,0 +1,56 @@
+use core_crypto::types::Hash;
+use core_types::acknowledgement::AcknowledgedTicket;
+use core_types::channels::ChannelEntry;
+use libp2p_identity::PeerId;
+use multiaddr::Multiaddr;
+use utils_types::primitives::{Address, Balance};
+
+/// Contains TX hash along with the Chain Event data.
+/// This could be used to pair up some events with `Action`
+#[derive(Debug, Clone, PartialEq)]
+pub struct SignificantChainEvent {
+    /// TX hash
+    pub tx_hash: Hash,
+    /// Chain event of interest
+    pub event_type: ChainEventType,
+}
+
+/// Status of a node in network registry.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum NetworkRegistryStatus {
+    /// Connections to the node are allowed.
+    Allowed,
+    /// Connections to the node are not allowed.
+    Denied,
+}
+
+/// Enumeration of HOPR chain events.
+#[derive(Debug, Clone, PartialEq)]
+pub enum ChainEventType {
+    /// Peer on-chain announcement event.
```

Comment:
Shouldn't these event types cover all events we have in our contracts to be future proof?

Reply:
This is more of a business logic event... than a chain one, maybe it could reside in general primitives.